dace/frontend/python/parser.py

```diff
@@ -282,50 +248,99 @@ def __call__(self, *args, **kwargs):
 
         # Invoke auto-optimization as necessary
         if Config.get_bool('optimizer', 'autooptimize') or self.auto_optimize:
-            from dace.transformation.auto import auto_optimize as autoopt
-            sdfg = autoopt.auto_optimize(sdfg, self.device)
+            sdfg = self._auto_optimize(sdfg)
 
         # Compile SDFG (note: this is done after symbol inference due to shape
         # altering transformations such as Vectorization)
         binaryobj = sdfg.compile()
 
+        self._cache = (argtypes, sdfg, binaryobj)
+
         # Call SDFG
         result = binaryobj(**kwargs)
 
-        # Enforce deletion and closure of compiled program
-        del binaryobj
-
         return result
 
+    def parse(self, *compilation_args, strict=None, save=False) -> SDFG:
+        """ 
+        Try to parse a DaceProgram object and return the `dace.SDFG` object
+        that corresponds to it.
+        :param function: DaceProgram object (obtained from the ``@dace.program``
+                        decorator).
+        :param compilation_args: Various compilation arguments e.g. dtypes.
+        :param strict: Whether to apply strict transformations or not (None
+                    uses configuration-defined value). 
```

Comment:
```suggestion
        :param strict: Whether to apply strict transformations or not (None
                       uses configuration-defined value). 
```

Reply:
I wish yapf could handle docstrings... There are [tools that can do this](https://github.com/myint/docformatter) (albeit separate), we could consider if it's worth adopting this at some point.

Reply:
yeah no idea why yapf doesn't do this.